src/stream/mod.rs

```diff
@@ -291,28 +328,34 @@ impl Stream {
         file: &mut File,
         channel: &Option<InternalSender>,
         count: usize,
+        content_length: Option<u64>,
     ) -> Result<usize> {
-        let res = self.get(url).await?;
-        self.write_stream_to_file(res.bytes_stream(), file, channel, count).await
+        let res = self.get(url, content_length).await?;
+        self.write_stream_to_file(res.bytes_stream(), file, channel, count)
+            .await
     }
 
     #[inline]
-    async fn get(&self, url: &url::Url) -> Result<reqwest::Response> {
+    async fn get(&self, url: &url::Url, content_length: Option<u64>) -> Result<reqwest::Response> {
         log::trace!("get: {}", url.as_str());
-        Ok(
+        let k = self.client.get(url.as_str());
+        Ok(if let Some(content_length) = content_length {
+            let k = k.header("Range", format!("bytes=0-{content_length}"));
```

Comment:
Unfortunately some streams have content length set as zero. One of them being the highest quality stream of the example video. This leads to zero bytes being returned by the server.

Reply:
I'll check on that but I don't know if there is a possible fix at least better than defaulting to old slow system...

Reply:
I'm pretty sure this is not idiomatic code (this basically being the first lines of Rust I've written outside of tutorials) - but here we go - we should probably fall back to not using the `Range` if the content length is zero.
```suggestion
            let k = if content_length > 0 {
                k.header("Range", format!("bytes=0-{content_length}"))
            } else {
                k
            };
```